### ISSUE TITLE
Active Target does not allow special or multiple targets

### ISSUE THREAD

**Issue by Trass3r** [microsoft/vscode-cmake-tools#1360]
Once the extension has built up its target list it's only possible to select one of the offered targets which are somehow filtered it seems.
So I can neither set e.g. ninja folder targets like src/all nor can I give multiple targets to build exactly what I need for my work.
If the target is not in the list it should just be accepted as-is and passed to ninja.

### Platform and Versions

- **Operating System**: Win10 Remote WSL Ubuntu 20.04
- **CMake Version**: 3.17
- **VSCode Version**: 1.47
- **CMake Tools Extension Version**: 1.4.1
- **Compiler/Toolchain**: clang-10

**Comment by andrewmcdonald-oxb** [comment]
I'm particularly interested in the filtering issue. We've also noticed it strips out `folder/all` named targets, which we often want to build. The only workaround we've found is to set CMake Tools not to configure on opening a folder, open the target picker which lets you type arbitrary text since it hasn't got a filter list yet, then press build. CMake Tools will pass your arbitrary string to CMake for building - so long as you've picked a `folder/all` target, it works.

Is there any chance of a quick resolution to it filtering out these targets?

**Comment by Trass3r** [comment]
The workaround I'm using is the _build using task_ setting.

**Comment by bobbrow** [comment]
Would either of you be able to provide a sample project that has the targets you're talking about? That would make it easier for one of us maintainers or someone in the community to pick up this feature request and implement it.

(This is not a statement that we'll get to it right away. It's just a way to make it more likely that someone can hit the ground running with this and implement it sooner.)

Thanks!

**Comment by Trass3r** [comment]
I guess any project with multiple targets generating Ninja files will do.

**Comment by andrewmcdonald-oxb** [comment]
That's right. The attached is a simple setup with CMake targets defined in two subfolders. CMake generates targets named `FolderA/all` and `FolderB/all` for conveniently building all targets in those subfolders, but they aren't picked up by CMake Tools in either the target picker or Project Outline pane.
https://cmake.org/cmake/help/latest/generator/Ninja.html

They are displayed if you generate then run `ninja -t targets all` from the command line, but there's a lot of other internal gubbins listed there too. I don't know what method CMake Tools uses to discover the target list.
https://ninja-build.org/manual.html#_extra_tools

[AllTest.zip](https://github.com/microsoft/vscode-cmake-tools/files/5497533/AllTest.zip)

**Comment by Trass3r** [comment]
Still valid?

**Comment by andrewmcdonald-oxb** [comment]
Yes, I'm still hopeful for this, as it's not available in the latest CMake Tools. @bobbrow is there enough info to understand what we'd like?

**Comment by gcampbell-msft** [comment]
@andrewmcdonald-oxb There does seem to be enough information to investigate this at our soonest availability, thanks for providing a sample project. Based on bandwidth, we will investigate this when we can. Also, we are happy to take open-source contributions for this feature!

**Comment by lmielick** [comment]
That would be a real time saver in large projects. On the UI design side it would make sense to add checkboxes in target picker similar to branch picker in a popular eamodio.gitlens extension.

**Comment by hannah-computer** [comment]
Maybe this should be split into two two issues: one for filtering, and one for selecting multiple targets?

- instead of a single selection, allow checkboxes in the picker, and pass through arbitrary inputs

<img width="323" height="218" alt="Image" src="https://github.com/user-attachments/assets/1d34b650-c67b-4fa8-a8ea-e6d539f0e16e" />

- (stretch) maybe allow shift-selecting multiple entries in the Project Outline browser's context menu, or "Add build target"?

<img width="284" height="239" alt="Image" src="https://github.com/user-attachments/assets/2e012700-9d27-4608-bdcd-38b70b3c2707" />

- I think CMake users wouldn't be too surprised if the [`cmake.buildTargetName`](https://github.com/microsoft/vscode-cmake-tools/blob/81fb88fb4c882bc1bc6528c7ed785e42328d1253/docs/cmake-settings.md#command-substitution) command substitution returned a list string like `firstTarget;secondTarget`. That's also understood by the `cmake --target` option.
